Project1/PartyBase.cpp

```diff
@@ -0,0 +1,57 @@
+#include "PartyBase.h"
+
+PartyBase::PartyBase() :
+	PartyBase("~NULL PARTY~", std::vector<UnitData>())
+{}
+
+PartyBase::PartyBase(std::string name) :
+	PartyBase(name, std::vector<UnitData>())
+{}
+
+PartyBase::PartyBase(std::string name, std::vector<UnitData> unit_data) :
+	_party_name(std::move(name)) {
+	for (UnitData& data : unit_data) {
+		insertUnit(data);
+	}
+}
+PartyBase::PartyBase(PartyData data) :
+	PartyBase(data.name, data.unit_data)
+{}
+
+PartyBase::const_iterator PartyBase::getPosition(const Unit & unit) const {
```

Comment:
What are these for? You give it a Unit and they return a iterator that points to that unit?